internal/service/advancedcluster/resource_test.go

```diff
@@ -2051,6 +2059,135 @@ func checkAdvancedDefaultWrite(name, writeConcern, tls string) resource.TestChec
 		pluralChecks...)
 }
 
+func TestAccAdvancedCluster_tls12to13CustomCipherUpdate(t *testing.T) {
+	var (
+		projectID, clusterName = acc.ProjectIDExecutionWithCluster(t, 4)
+		resourceName           = "mongodbatlas_advanced_cluster.test"
+		processArgsTLS12       = &admin.ClusterDescriptionProcessArgs20240805{
+			TlsCipherConfigMode:            conversion.StringPtr("CUSTOM"),
+			CustomOpensslCipherConfigTls12: conversion.Pointer([]string{"TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256"}),
+			MinimumEnabledTlsProtocol:      conversion.StringPtr("TLS1_2"),
+		}
+		processArgsTLS13 = &admin.ClusterDescriptionProcessArgs20240805{
+			TlsCipherConfigMode:            conversion.StringPtr("CUSTOM"),
+			CustomOpensslCipherConfigTls12: conversion.Pointer([]string{}),
+			CustomOpensslCipherConfigTls13: conversion.Pointer([]string{"TLS_AES_128_GCM_SHA256"}),
+			MinimumEnabledTlsProtocol:      conversion.StringPtr("TLS1_3"),
+		}
+
+		processArgsTLS12Revert = &admin.ClusterDescriptionProcessArgs20240805{
+			TlsCipherConfigMode:            conversion.StringPtr("CUSTOM"),
+			CustomOpensslCipherConfigTls12: conversion.Pointer([]string{"TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256"}),
+			CustomOpensslCipherConfigTls13: conversion.Pointer([]string{}),
+			MinimumEnabledTlsProtocol:      conversion.StringPtr("TLS1_2"),
+		}
```

Comment:
```suggestion
		processArgsTLS12       = &admin.ClusterDescriptionProcessArgs20240805{
			TlsCipherConfigMode:            conversion.StringPtr("CUSTOM"),
			CustomOpensslCipherConfigTls12: conversion.Pointer([]string{"TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256"}),
			MinimumEnabledTlsProtocol:      conversion.StringPtr("TLS1_2"),
		}
		processArgsTLS13 = &admin.ClusterDescriptionProcessArgs20240805{
			TlsCipherConfigMode:            conversion.StringPtr("CUSTOM"),
			CustomOpensslCipherConfigTls13: conversion.Pointer([]string{"TLS_AES_128_GCM_SHA256"}),
			MinimumEnabledTlsProtocol:      conversion.StringPtr("TLS1_3"),
		}

```

I think the best test is to have only defined what you need in each step, and use again processArgsTLS12 in the 3rd step, i.e. don't declare the empty lists, the code should take care of it

Reply:
Sounds good, removed unnecessary fields plus extra check on [a7982b8](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/3912/commits/a7982b81a127fcd4b5396b7d89e3b265690cbcd6), plus addressed the code behaviour that should take care of empty lists on [3e80331](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/3912/commits/3e80331cfc742250962d5b9a1783b091b0a70ca0)